### ISSUE TITLE
Deploying to existing cluster hal-deploy-apply error

### ISSUE THREAD

**Issue by selfeky** [GoogleCloudPlatform/spinnaker-for-gcp#88]
Having an issue running on an existing cluster
Stuck on:
`Waiting on job hal-deploy-apply to complete...`

Looking at the logs, I'm seeing:
`Failed to ensure the required bucket "spinnaker-plv5i7js7ndopdyyggqv-1564919957" exists:`
`com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden`
`Insufficient Permission`

This bucket doesn't exist and was not created by the setup script.

Can't find where this is set.

Tried cleaning up with a generated clean up script as well as deleting both spinnaker and halyard namespaces but the error is the same, with the same bucket name.

Thanks

**Comment by duftler** [comment]
This looks to be related to the service account not having the needed permissions to create the bucket. Please ensure that the service account used by the nodes in your gke nodepool is the same one that was declared in your properties file and that it has the correct IAM permissions.

Have you gotten this to work all the way through on a new cluster? I'd recommend trying that first before trying to deploy on an existing cluster. This way you'll have a close-by setup to compare your new setup against.

**Comment by demobox** [comment]
@selfeky A quick follow-up ping here - were you able to try duftler's suggestion?

**Comment by selfeky** [comment]
@demobox Yes! I did.

Going to close the issue now and open one with proper description once I have everything ready.

Thanks

**Comment by demobox** [comment]
> Yes! I did.

Glad to hear - looking forward to the follow-up issue!
